src/utils/clients.py

```
@@ -359,13 +373,42 @@ async def honcho_llm_call_inner(
+                    repaired_json = validate_and_repair_json(raw_content)
+
+                    # Parse JSON and validate against model
+                    parsed_content = response_model.model_validate_json(repaired_json)
+
+                    return HonchoLLMCallResponse(
+                        content=parsed_content,
+                        output_tokens=usage.output_tokens if usage else 0,  # pyright: ignore
+                        finish_reasons=[stop_reason] if stop_reason else [],
+                    )
+                except (json.JSONDecodeError, ValidationError) as e:
+                    logger.error(f"Failed to parse Anthropic response as {response_model.__name__}: {e}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)